core/src/marine_core.rs

```diff
@@ -61,50 +61,73 @@ pub struct MarineCore<WB: WasmBackend> {
 }
 
 impl<WB: WasmBackend> MarineCore<WB> {
-    pub fn new(config: MarineCoreConfig) -> MResult<Self> {
-        let wasm_backend = WB::new()?;
-        let mut store = <WB as WasmBackend>::Store::new(&wasm_backend);
+    pub fn new(config: MarineCoreConfig<WB>) -> MResult<Self> {
+        let mut store = <WB as WasmBackend>::Store::new(&config.wasm_backend);
         store.set_total_memory_limit(config.total_memory_limit);
         Ok(Self {
             modules: HashMap::new(),
-            wasm_backend,
+            wasm_backend: config.wasm_backend,
             store: RefCell::new(store),
         })
     }
 
     /// Invoke a function of a module inside Marine by given function name with given arguments.
-    pub fn call(
+    pub async fn call(
```

Comment:
What the difference between `call` and `call_async`? The same for corresponding functions on the `MModule` level.

Reply:
call_async ones should be removed, they are obsolete

Reply:
Hmm. I think it is better to rename everything to `call_async` now and add plain `call`  if we decide to add a sync variant later